src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -259,7 +263,31 @@ private static string FindBestVersionMatch(VersionRange versionRange, IEnumerabl
                 return version;
             }).Where(v => v != null);
 
-            return bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault()?.ToString();
+            var matchingVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+
+            if (bundleId != ScriptConstants.DefaultExtensionBundleId)
+            {
+                return matchingVersion?.ToString();
+            }
+
+            // Check to see if there is a max bundle version set via hosting configuration, if yes then use that instead of the one
+            // available on VM or local machine
+            if (!string.IsNullOrEmpty(configOption.MaximumBundleV3Version))
+            {
+                var maximumBundleV3Version = NuGetVersion.Parse(configOption.MaximumBundleV3Version);
+                matchingVersion = matchingVersion?.Major == ScriptConstants.ExtensionBundleV3MajorVersion && matchingVersion > maximumBundleV3Version
+                                ? maximumBundleV3Version
```

Comment:
Just to confirm my understanding. I see we return `maximumBundleV3Version` when `matchingVersion > maximumBundleV3Version`. Did you mean to return that? Or did you wanted to return matchingVersion?

Should we add comments about the logic? (Why we do this)?

Reply:
Yes your understanding is correct. We would use maxBundleVersion in case one is set via hosting configuration. I have some comments added explaining this. However there are a couple more things i can add.